lms/djangoapps/courseware/courses.py

```diff
@@ -115,6 +115,20 @@ def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=
     return course
 
 
+def get_discussion_course_or_404(course_key, user):
+    """
+    Given a course_key, look up the corresponding course descriptor,
+    check that the user has the access to perform the specified action
+    on the course, discussion is enabled for the course and return the descriptor
+
+    Raises a 404 if the course_key is invalid, the user doesn't have access or discussion is disabled
```

Comment:
Nit: Docstrings should follow these conventions, https://github.com/edx/edx-platform/wiki/Python-Guidelines

search for `Here's how you write documentation in "Google Style":`